qiita_db/reference.py

```diff
@@ -113,54 +115,62 @@ def exists(cls, name, version):
-        return conn_handler.execute_fetchone(
-            "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
-            "reference_name=%s AND reference_version=%s)".format(cls._table),
-            (name, version))[0]
+        with TRN:
+            sql = """SELECT EXISTS(
+                        SELECT * FROM qiita.{0}
+                        WHERE reference_name=%s
+                            AND reference_version=%s)""".format(cls._table)
+            TRN.add(sql, [name, version])
+            return TRN.execute_fetchlast()
 
     @property
     def name(self):
-        conn_handler = SQLConnectionHandler()
-        return conn_handler.execute_fetchone(
-            "SELECT reference_name FROM qiita.{0} WHERE "
-            "reference_id = %s".format(self._table), (self._id,))[0]
-        _, basefp = get_mountpoint('reference')[0]
+        with TRN:
+            sql = """SELECT reference_name FROM qiita.{0}
+                     WHERE reference_id = %s""".format(self._table)
+            TRN.add(sql, [self._id])
+            return TRN.execute_fetchlast()
 
     @property
     def version(self):
-        conn_handler = SQLConnectionHandler()
-        return conn_handler.execute_fetchone(
-            "SELECT reference_version FROM qiita.{0} WHERE "
-            "reference_id = %s".format(self._table), (self._id,))[0]
-        _, basefp = get_mountpoint('reference')[0]
+        with TRN:
+            sql = """SELECT reference_version FROM qiita.{0}
+                     WHERE reference_id = %s""".format(self._table)
+            TRN.add(sql, [self._id])
+            return TRN.execute_fetchlast()
 
     @property
     def sequence_fp(self):
-        conn_handler = SQLConnectionHandler()
-        rel_path = conn_handler.execute_fetchone(
-            "SELECT f.filepath FROM qiita.filepath f JOIN qiita.{0} r ON "
-            "r.sequence_filepath=f.filepath_id WHERE "
-            "r.reference_id=%s".format(self._table), (self._id,))[0]
-        _, basefp = get_mountpoint('reference')[0]
-        return join(basefp, rel_path)
+        with TRN:
+            sql = """SELECT f.filepath
+                     FROM qiita.filepath f
+                        JOIN qiita.{0} r ON r.sequence_filepath=f.filepath_id
+                     WHERE r.reference_id=%s""".format(self._table)
+            TRN.add(sql, [self._id])
+            rel_path = TRN.execute_fetchlast()
+            _, basefp = get_mountpoint('reference')[0]
+            return join(basefp, rel_path)
 
     @property
     def taxonomy_fp(self):
-        conn_handler = SQLConnectionHandler()
-        rel_path = conn_handler.execute_fetchone(
-            "SELECT f.filepath FROM qiita.filepath f JOIN qiita.{0} r ON "
-            "r.taxonomy_filepath=f.filepath_id WHERE "
-            "r.reference_id=%s".format(self._table), (self._id,))[0]
-        _, basefp = get_mountpoint('reference')[0]
-        return join(basefp, rel_path)
+        with TRN:
+            sql = """SELECT f.filepath
```

Comment:
Please read carefully the SQL query... using does not work as the column does not have the same name in both tables.